File: level_1/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
    var answer = [];
    const rank = {1 : 6, 2 : 5, 3 : 4, 4 : 3, 5 : 2};
    let my = lottos.filter(x => !win_nums.includes(x));
    let win = win_nums.filter(x => !lottos.includes(x));

    let count = lottos.length - my.length;
    let count_zero = 0;
    my.forEach(x => {
        if(x === 0) count_zero++
    });
        
    const good = count + count_zero;
    const bad = count;
    
    for(let i = 1; i < 6; i++){
        if(rank[i] == good) answer.push(i);
        if(rank[i] == bad) answer.push(i);
    }
    if(answer.length < 2) answer.push(6);
    if(answer.length < 2) answer.push(6);

    return answer;
}

File: level_1/로또의 최고 순위와 최저 순위.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int *solution(int lottos[], size_t lottos_len, int win_nums[], size_t win_nums_len)
{
    int result[7] = {6, 6, 5, 4, 3, 2, 1};
    int *answer = (int *)malloc(sizeof(int) * 2);
    int same_cnt = 0;
    int zero_cnt = 0;
    for (int i = 0; i < lottos_len; i++)
    {
        if (lottos[i] == 0)
        {
            zero_cnt++;
            continue;
        }
        for (int j = 0; j < win_nums_len; j++)
        {
            if (lottos[i] == win_nums[j])
            {
                same_cnt++;
                break;
            }
        }
    }
    answer[0] = result[same_cnt + zero_cnt];
    answer[1] = result[same_cnt];
    return answer;
}